cmd/abapEnvironmentCheckoutBranch.go

```diff
@@ -74,38 +76,32 @@ func runAbapEnvironmentCheckoutBranch(options *abapEnvironmentCheckoutBranchOpti
 	client.SetOptions(clientOptions)
 	pollIntervall := com.GetPollIntervall()
 
-	log.Entry().Infof("Starting to switch branch to branch '%v' on repository '%v'", options.BranchName, options.RepositoryName)
-	log.Entry().Info("--------------------------------")
-	log.Entry().Info("Start checkout branch: " + options.BranchName)
-	log.Entry().Info("--------------------------------")
+	checkCheckoutBranchRepositoryConfiguration(*options)
 
-	// Triggering the Checkout of the repository into the ABAP Environment system
-	uriConnectionDetails, errorTriggerCheckout := triggerCheckout(options.RepositoryName, options.BranchName, connectionDetails, client)
-	if errorTriggerCheckout != nil {
-		log.Entry().WithError(errorTriggerCheckout).Fatal("Checkout of " + options.BranchName + " for software component " + options.RepositoryName + " failed on the ABAP System")
+	if len(options.RepositoryNamesFiles) > 0 {
+		err = checkoutBranchesFromFileConfig(options.RepositoryNamesFiles, connectionDetails, client, pollIntervall)
```

Comment:
With those two functions "checkoutBranchesFromFileConfig" and "checkoutBranchFromConfig" there are a lot of duplications, which IMO leads to not well maintainable code. And essentially, those two functions are doing the same thing. 

A better approach would be to combine both sources of repositories into one list/map and then loop over this one list. (see https://github.com/SAP/jenkins-library/blob/master/cmd/abapEnvironmentCloneGitRepo.go)